File: LeetcodeJava/src/Num242.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Num242 {
    public static void main(String[] args) {
//        String s = "anagram";
//        String t = "nagaram";

//        String s = "rat";
//        String t = "car";

        String s = "a";
        String t = "ab";
        System.out.println(isAnagram(s, t));
    }

    public static boolean isAnagram(String s, String t) {
        char[] sChar = s.toCharArray();
        char[] tChar = t.toCharArray();
        Arrays.sort(sChar);
        Arrays.sort(tChar);
        String sString = new String(sChar);
        String tString = new String(tChar);
        return sString.equals(tString);
    }
}

File: Leetcode/src/main/kotlin/Num242.kt
import java.util.*
import kotlin.math.max

fun main(args: Array<String>) {
//        String s = "anagram";
//        String t = "nagaram";

//        String s = "rat";
//        String t = "car";
    val s = "a"
    val t = "ab"

//        String s = "ab";
//        String t = "a";
    println(isAnagram(s, t))
}

fun isAnagram(s: String, t: String): Boolean {
    val sChar = s.toCharArray()
    val tChar = t.toCharArray()
    Arrays.sort(sChar)
    Arrays.sort(tChar)
    val n = max(sChar.size, tChar.size)
    for (i in 0 until n) {
        if ((sChar[i] ?: 0) != (tChar[i] ?: 0)) {
            return false
        }
    }
    return true
}